src/main/java/de/dataelementhub/model/service/ImportService.java

```diff
@@ -0,0 +1,240 @@
+   **/
+  @Async
+  public void importService(
+      List<MultipartFile> file, String importDirectory, int userId,
+      int importId, String timestamp) {
+    try (CloseableDSLContext ctx = ResourceManager.getDslContext()) {
+      ImportHandler.importFiles(ctx, file, importDirectory,
+          userId, importId, timestamp);
+    }
+  }
+
+  /**
+   * Convert stagedElements to Drafts.
+   **/
+  @Async
+  public void convertToDraft(
+      List<String> stagedElementsIds, int userId, int importId) throws
+      IllegalAccessException {
+    try (CloseableDSLContext ctx = ResourceManager.getDslContext()) {
+      if (!importExists(ctx, importId)) {
+        throw new NoSuchElementException();
+      }
+      if (!importAccessGranted(ctx, importId, userId)) {
+        throw new IllegalAccessException();
+      }
+      ImportHandler.convertToDrafts(ctx, importId, userId, stagedElementsIds);
+    }
+  }
+
+  /** Check user grants then delete staged import. */
+  public void deleteStagedImport(int userId, int importId) throws IllegalAccessException,
+      NoSuchElementException {
+    try (CloseableDSLContext ctx = ResourceManager.getDslContext()) {
+      if (!importExists(ctx, importId)) {
+        throw new NoSuchElementException();
+      }
+      Integer namespaceIdentifier = ctx.select().from(IMPORT).leftJoin(SCOPED_IDENTIFIER)
+          .on(IMPORT.NAMESPACE_ID.eq(SCOPED_IDENTIFIER.ELEMENT_ID))
+          .where(IMPORT.ID.eq(importId)).fetchOne().getValue(SCOPED_IDENTIFIER.IDENTIFIER);
+      if (importAccessGranted(ctx, importId, userId)) {
+        ctx.deleteFrom(IMPORT).where(IMPORT.ID.eq(importId)).execute();
+        ctx.deleteFrom(STAGING).where(STAGING.IMPORT_ID.eq(importId)).execute();
+      } else {
+        throw new IllegalAccessException();
+      }
+    }
+  }
+
+  /** Generate importId. */
+  public int generateImportId(String namespaceUrn, int userId, List<MultipartFile> files,
+      String importDirectory, String timestamp) throws IOException {
+    try (CloseableDSLContext ctx = ResourceManager.getDslContext()) {
+      String destination = importDirectory + "/" + userId + "/" + timestamp;
+      new File(importDirectory + "/" + userId).mkdir();
+      new File(destination).mkdir();
+      for (MultipartFile file : files) {
+        Path fileNameAndPath = Paths.get(destination, file.getOriginalFilename());
+        Files.write(fileNameAndPath, file.getBytes());
+      }
+      int namespaceId = IdentificationHandler
+          .getScopedIdentifier(ctx, namespaceUrn).getNamespaceId();
+      ImportRecord importIdRecord = ctx.insertInto(IMPORT)
+          .set(IMPORT.NAMESPACE_ID, namespaceId)
+          .set(IMPORT.STATUS, ProcessStatus.PROCESSING)
+          .set(IMPORT.CREATED_BY, userId)
+          .returning(IMPORT.ID)
+          .fetchOne();
+      return importIdRecord != null ? importIdRecord.getId() : -1;
+    }
+  }
+
+  /** Returns all Imports. */
+  public List<ImportInfo> allImports(int userId) {
```

Comment:
imho the method name should also contain the action for the imports..."readAllImports" or "getAllImports" or "listAllImports"...or something like that

Reply:
renamed to listAllImports